### ISSUE TITLE
torchtext.utils.unicode_csv_reader` func reads multiple lines which collapse into a single line

### ISSUE THREAD

**Issue by zhangguanheng66** [pytorch/text#1081]
## 🐛 Bug
**Describe the bug**
`torchtext.utils.unicode_csv_reader` doesn't read multiple lines properly. Sometimes, the multiple lines collapse into a single line.

**To Reproduce**
1. download the text file
```
wget https://pytorch.s3.amazonaws.com/models/text/train.de-en.en
```
2. Run the following code snippet and see the difference
```
import io
from torchtext.utils import unicode_csv_reader
with io.open('train.de-en.en', encoding="utf8") as f:
    reader = unicode_csv_reader(f)
    print('unicode_csv_reader:')
    for row in reader:
        print(row)
print('\n')
with io.open('train.de-en.en', encoding="utf8") as f:
    print('built-in text reader:')
    for row in f:
        print(row)
```

See different results as follows
```
unicode_csv_reader:
['I said "play', " all right.\nIn case some of you forgot what play is, this is what play looks like.\nBabies play, kids play, adults play.\nWe don't value play.\n"]


built-in text reader:
I said "play," all right.

In case some of you forgot what play is, this is what play looks like.

Babies play, kids play, adults play.

We don't value play.
```

**Comment by cpuhrsch** [comment]
What is it within unicode_csv_reader that causes this issue?

**Comment by zhangguanheng66** [comment]
It is the `csv` package within `unicode_csv_reader` func that collapse the multiple lines into a single line. To reproduce the error:
```
import csv
from torchtext.utils import unicode_csv_reader
with io.open('train.de-en.en', encoding="utf8") as f:
    reader = csv.reader(f)
    print('csv_reader:')
    for row in reader:
        print(row)
```

**Comment by cpuhrsch** [comment]
Could this maybe be related to the text encoding used for the input file?

**Comment by zhangguanheng66** [comment]
> Could this maybe be related to the text encoding used for the input file? If this is due to the csv package, this is a good candidate for https://bugs.python.org/. You can use [enca](https://linux.die.net/man/1/enconv) to figure out the encoding of a file. Maybe the input file isn't compatible with the utf8 encoding option used by io.open.
> 
> But really the difference is that you're treating the file as a csv file vs. not treating it as such. That means a different delimiter is applied. Maybe the input file is meant to be used as a csv file and the collapsing is on purpose? We should double check the source of the dataset. Also [check the defaults of the csv reader](https://docs.python.org/3/library/csv.html#csv.reader) to see what the default delimiter is.
> 
> In general, if the input file is not a csv file the raw dataset readers shouldn't be treating them as such. It would be good to know and flag how we ended up doing so.

Yup. I search the raw datasets in torchtext, and only text classification datasets and translation datasets are using `unicode_csv_reader`.
